deploy/setup/scripts/web-config.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+set -e
+
+echo
+echo "🔧 Starting web-based configuration wizard..."
+echo
+
+# --- Expected env vars ---
+DEPLOY_DIR="${DEPLOY_DIR:?DEPLOY_DIR not set}"
+LOG_PATH="${LOG_PATH:-/tmp/wbs-deploy-setup.log}"
+VERBOSE="${VERBOSE:-false}"
+LOCALHOST="${LOCALHOST:-false}"
+CLOUD_INIT="${CLOUD_INIT:-false}"
+CERT_EMAIL="${CERT_EMAIL:-wbs-setup@wikimedia.de}"
+
+# -- Script specific env vars --
+SETUP_PORT=8888
+SETUP_DIR="$DEPLOY_DIR/setup"
+CERTS_DIR="$SETUP_DIR/certs"
+VIEWS_DIR="$SETUP_DIR/views"
+
+# -- Setup logging --
+# shellcheck disable=SC1091
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/_logging.sh"
+
+generate_cert_for_setup_webserver() {
+  log "Generating TLS certificate for setup page..."
+
+  if [ "$LOCALHOST" = true ]; then
+    SETUP_HOST="localhost"
+  else
+    PUBLIC_IP=$(curl --silent --show-error --fail https://api.ipify.org)
+    if [[ -z "$SETUP_HOST" ]]; then
+      if [[ "$CLOUD_INIT" = true ]]; then
+        SETUP_HOST="$PUBLIC_IP.nip.io"
+      else
+        SETUP_SUBDOMAIN="wbs-setup-$(LC_ALL=C tr -dc 'a-z0-9' </dev/urandom | head -c 6)"
+        SETUP_HOST="$SETUP_SUBDOMAIN.$PUBLIC_IP.nip.io"
+      fi
+    fi
+  fi
+
+  log_cmd "mkdir -p $SETUP_DIR/letsencrypt $CERTS_DIR"
+  log "Using domain: $SETUP_HOST"
+
+  log_cmd "docker pull certbot/certbot"
+
+  if log_cmd "docker run --rm \
+    -v $SETUP_DIR/letsencrypt:/etc/letsencrypt \
+    -v $CERTS_DIR:/certs \
+    -p 80:80 \
+    certbot/certbot certonly \
```

Comment:
Do we want to lock the version here? WDYT?

Reply:
Yes, and I've went ahead and pinned it now to v4.2.0 (they don't have sematic version series pins like us :)